webapp/channels/src/components/channel_view/channel_view.tsx

```diff
@@ -86,6 +88,16 @@ export default class ChannelView extends React.PureComponent<Props, State> {
     };
 
     componentDidUpdate(prevProps: Props) {
+        // TODO: debounce
+        if (prevProps.channelId !== this.props.channelId) {
+            const cb = (msg: any) => {
+                // retrying only once for now.
+                if (msg.status !== 'OK') {
+                    WebSocketClient.updateActiveChannel(this.props.channelId);
+                }
+            };
+            WebSocketClient.updateActiveChannel(this.props.channelId, cb);
```

Comment:
should we wait out here a bit before firing the update?

Reply:
Any reason for doing that?

Reply:
i was thinking of case where user switches channels quickly, will that be a problem?

Reply:
Yes debouncing is a TODO item which we need to do.